superstore/src/commonMain/kotlin/org/mobilenativefoundation/store/superstore5/impl/RealSuperstore.kt

```diff
@@ -0,0 +1,87 @@
+package org.mobilenativefoundation.store.superstore5.impl
+
+
+import kotlinx.coroutines.channels.ProducerScope
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.channelFlow
+import org.mobilenativefoundation.store.store5.Store
+import org.mobilenativefoundation.store.store5.StoreReadRequest
+import org.mobilenativefoundation.store.store5.StoreReadResponse
+import org.mobilenativefoundation.store.store5.StoreReadResponseOrigin
+import org.mobilenativefoundation.store.superstore5.Superstore
+import org.mobilenativefoundation.store.superstore5.SuperstoreResponse
+import org.mobilenativefoundation.store.superstore5.SuperstoreResponseOrigin
+import org.mobilenativefoundation.store.superstore5.Warehouse
+
+/**
+ * Coordinates [Store] and [Warehouse].
+ * Tries to fetch data through [Store].
+ * On failure, fetches data using [Warehouse] until a success.
+ */
+class RealSuperstore<Key : Any, Output : Any>(
+    private val store: Store<Key, Output>,
+    private val warehouses: List<Warehouse<Key, Output>>,
+) : Superstore<Key, Output> {
+    override fun get(key: Key, fresh: Boolean, refresh: Boolean): Flow<SuperstoreResponse<Output>> {
+        val request =
+            if (fresh) StoreReadRequest.fresh(key) else StoreReadRequest.cached(key, refresh)
+        return channelFlow {
+            store.stream(request).collect {
+                try {
+                    when (it) {
+                        is StoreReadResponse.Data -> useStore(
+                            it.requireData(),
+                            it.origin.toSuperstoreResponseOrigin()
+                        )
+
+                        is StoreReadResponse.Loading -> send(SuperstoreResponse.Loading)
+                        is StoreReadResponse.NoNewData -> {}
+                        is StoreReadResponse.Error.Exception -> useWarehouse(key)
+                        is StoreReadResponse.Error.Message -> useWarehouse(key)
+                    }
+                } catch (error: Throwable) {
+                    useWarehouse(key)
+                }
+            }
+        }
+    }
+
```

Comment:
Just for parity to `Store` should we consider expose `get(key)` and `fresh(key)` instead of passing two booleans for keeping the same style?

Reply:
Yeah good idea 👍🏽

Reply:
Changed to use `StoreReadRequest`